### PR TITLE
[LM-9641] Image Cropping

### PR DESCRIPTION
**Pull request by jaixchaudhary**
## Description

**Features**
- Image Cropping

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [x] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: ChatSX/screens/ImageCrop/index.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+          <CropView
+            sourceUrl={uri}
+            style={styles.cropView}
+            ref={cropViewRef}
+            onImageCrop={res => {
+              let croppedImage = {...res};
+              if (Platform.OS === 'android') {
+                if (!croppedImage?.uri) return;
+                croppedImage.uri = `file://${croppedImage?.uri}`;
+              }
```
Comment (ishaan1607) [inline ChatSX/screens/ImageCrop/index.tsx:61]: what will happen for iOS?

File: ChatSX/screens/ImageCrop/index.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+          style={styles.item}
+          onPress={() => {
+            cropViewRef?.current?.rotateImage(true);
+          }}>
+          <Image
+            source={require('../../assets/images/rotate_icon3x.png')}
+            style={styles.rotateIcon}
+          />
+          {/* <Text>Rotate</Text> */}
+        </TouchableOpacity>
```
Comment (ishaan1607) [inline ChatSX/screens/ImageCrop/index.tsx:87]: remove commented code

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: ChatSX/navigation/SwitchComponent/models/RootStackParamsList.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  ViewParticipants: undefined;
+  AddParticipants: undefined;
+  DMAllMembers: undefined;
+  FileUpload: undefined;
+  VideoPlayer: undefined;
+  CarouselScreen: undefined;
+  PollResult: undefined;
+  CreatePollScreen: undefined;
+  ImageCropScreen: {uri: string; fileName: string};
+};
```
[the remark this answers is not in the document]
Comment (jaixchaudhary) [reply ChatSX/navigation/SwitchComponent/models/RootStackParamsList.ts:16]: For passing component props like(navigation, and route) that have passed through NavigationContainer. For previously built components we have to handle. But for now, we have used props type only in ImageCropScreen.

File: ChatSX/screens/ImageCrop/index.tsx (hunk 1 above)
Comment (jaixchaudhary) [reply ChatSX/screens/ImageCrop/index.tsx:61]: For iOS `file://` is already prefixed by library itself

**Review by ishaan1607 — APPROVED** [review]

**Pull request merged**